### ISSUE TITLE
MD032: List continuation text in a nested block quote can be seen as a separate list

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#519]
### rumdl version

0.1.45

### Installation method

mise

### Bug description

Rule MD032 (https://github.com/rvben/rumdl/blob/a304d288c836b694e792d329a782db3e3165858c/docs/md032.md) helps ensure that separate lists are properly separated.

However, the rule can catch examples of list continuation text in a nested block quote that shouldn't be seen as separate.

### Steps to reproduce

1. Create a Markdown file with this content:
   ```markdown
   ---
   title: Heading
   ---
   
   Introductory text:
   
   - Level 1 item:
     - Level 2 item:
       > - Level 3 item.
       >   List continuation text.
       > - Level 3 item.
       >   List continuation text.
       >   More list continuation text.
   ```
1. Run `rumdl check` against the Markdown file.

### Expected behavior

Given the scenario above, I'd expect no violations.

### Actual behavior

Given the scenario above, I get the following:

```shell
test.md:12:1: [MD032] List should be followed by blank line [*]
```

### Configuration

None.

### Operating system

macOS

### Additional context

There is a few ways to change the Markdown in the scenario to stop `rumdl` flagging any problems. This Markdown sample is parsed fine:

```markdown
---
title: Heading
---

Introductory text:

- Level 1 item:
  - Level 2 item:
    > - Level 3 item.
    > - Level 3 item.
    >   List continuation text.
    >   More list continuation text.
```

Also, this is parsed fine:

```markdown
---
title: Heading
---

Introductory text:

- Level 1 item:
  - Level 2 item:
    > - Level 3 item.
    >   List continuation text.
    > - Level 3 item.
    >   List continuation text.
```

**Comment by rvben** [comment]
Fixed in v0.1.46. The blockquote prefix comparison now trims whitespace before matching, and a new `indent_after_blockquote()` helper correctly computes effective indent for continuation lines inside nested blockquotes. Thanks for reporting @eread!
